lintcheck/src/json.rs

```diff
@@ -130,9 +143,65 @@ pub(crate) fn diff(old_path: &Path, new_path: &Path) {
     println!();
     println!();
 
-    print_warnings("Added", &added);
-    print_warnings("Removed", &removed);
-    print_changed_diff(&changed);
+    print_warnings("Added", added);
+    print_warnings("Removed", removed);
+    print_changed_diff(changed);
```

Comment:
I think we should truncate them before here and print the message near the top, things in the middle of big lists tend to get lost

Reply:
Do you mean that the info should be printed directly after the heading or before any headings begin?

My thought was that this is not really that important.

This brings me an idea:

What do you think about splitting the `added/removed/changed` thing into lint sections? I imagine something like this:

> | lint | added | removed | changed |
> | --- | --- |--- |--- |
> | clippy::cool_lint | 12 | 21 | 2 |
> | clippy::too_few_cats | 2190 (log truncated) | 0 | 4 |
> 
> ## `clippy::cool_lint`
> ### Added
> ### Removed
> ### Changed
>
> ## `clippy::too_few_cats`
> ### Added
> 1990 warnings have been truncated.
> ### Removed
> ### Changed

For a single lint, it would be very similar to the current output, but large refactorings would benefit from this

Reply:
Yeah I meant just before `Added`

I like the table and splitting it per lint more though 👍